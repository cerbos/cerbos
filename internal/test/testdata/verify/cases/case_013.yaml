description: Valid test suite with lenient scopes
