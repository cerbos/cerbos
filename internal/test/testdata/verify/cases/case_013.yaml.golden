{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_PASSED",
        "testsCount": 24,
        "resultCounts": [
          {
            "result": "RESULT_PASSED",
            "count": 24
          }
        ]
      },
      "testCases": [
        {
          "name": "John's request",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "john_equipment_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault#public-view",
                              "val": {
                                "formatted_string": "id:john",
                                "id": "john",
                                "keys": "XX125",
                                "some_bool": true,
                                "some_list": [
                                  "foo",
                                  "bar"
                                ],
                                "something_nested": {
                                  "nested_bool": false,
                                  "nested_formatted_string": "id:john",
                                  "nested_list": [
                                    "nest_foo",
                                    1.01
                                  ],
                                  "nested_str": "foo"
                                }
                              }
                            }
                          ]
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault/acme#rule-001",
                              "val": "create_allowed:john"
                            }
                          ]
                        }
                      }
                    }
                  ]
                },
                {
                  "name": "john_pending_equipment_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault#public-view",
                              "val": {
                                "formatted_string": "id:john",
                                "id": "john",
                                "keys": "XX125",
                                "some_bool": true,
                                "some_list": [
                                  "foo",
                                  "bar"
                                ],
                                "something_nested": {
                                  "nested_bool": false,
                                  "nested_formatted_string": "id:john",
                                  "nested_list": [
                                    "nest_foo",
                                    1.01
                                  ],
                                  "nested_str": "foo"
                                }
                              }
                            }
                          ]
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault/acme#rule-001",
                              "val": "create_allowed:john"
                            }
                          ]
                        }
                      }
                    }
                  ]
                }
              ]
            },
            {
              "name": "bev",
              "resources": [
                {
                  "name": "john_equipment_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault#public-view",
                              "val": {
                                "formatted_string": "id:bev",
                                "id": "bev",
                                "keys": "XX125",
                                "some_bool": true,
                                "some_list": [
                                  "foo",
                                  "bar"
                                ],
                                "something_nested": {
                                  "nested_bool": false,
                                  "nested_formatted_string": "id:bev",
                                  "nested_list": [
                                    "nest_foo",
                                    1.01
                                  ],
                                  "nested_str": "foo"
                                }
                              }
                            }
                          ]
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault#rule-002",
                              "val": "condition_not_met"
                            }
                          ]
                        }
                      }
                    },
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault/acme#rule-001",
                              "val": "create_allowed:bev"
                            }
                          ]
                        }
                      }
                    }
                  ]
                },
                {
                  "name": "john_pending_equipment_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault#public-view",
                              "val": {
                                "formatted_string": "id:bev",
                                "id": "bev",
                                "keys": "XX125",
                                "some_bool": true,
                                "some_list": [
                                  "foo",
                                  "bar"
                                ],
                                "something_nested": {
                                  "nested_bool": false,
                                  "nested_formatted_string": "id:bev",
                                  "nested_list": [
                                    "nest_foo",
                                    1.01
                                  ],
                                  "nested_str": "foo"
                                }
                              }
                            }
                          ]
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault#rule-002",
                              "val": "approval_status:bev:PENDING_APPROVAL"
                            }
                          ]
                        }
                      }
                    },
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault/acme#rule-001",
                              "val": "create_allowed:bev"
                            }
                          ]
                        }
                      }
                    }
                  ]
                }
              ]
            },
            {
              "name": "terry_tibbs",
              "resources": [
                {
                  "name": "john_equipment_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault#public-view",
                              "val": {
                                "formatted_string": "id:terry_tibbs",
                                "id": "terry_tibbs",
                                "keys": "XX125",
                                "some_bool": true,
                                "some_list": [
                                  "foo",
                                  "bar"
                                ],
                                "something_nested": {
                                  "nested_bool": false,
                                  "nested_formatted_string": "id:terry_tibbs",
                                  "nested_list": [
                                    "nest_foo",
                                    1.01
                                  ],
                                  "nested_str": "foo"
                                }
                              }
                            }
                          ]
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "principal.terry_tibbs.vdefault#create-rule",
                              "val": [
                                "foo",
                                [
                                  "bar",
                                  true
                                ]
                              ]
                            }
                          ]
                        }
                      }
                    }
                  ]
                },
                {
                  "name": "john_pending_equipment_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "resource.equipment_request.vdefault#public-view",
                              "val": {
                                "formatted_string": "id:terry_tibbs",
                                "id": "terry_tibbs",
                                "keys": "XX125",
                                "some_bool": true,
                                "some_list": [
                                  "foo",
                                  "bar"
                                ],
                                "something_nested": {
                                  "nested_bool": false,
                                  "nested_formatted_string": "id:terry_tibbs",
                                  "nested_list": [
                                    "nest_foo",
                                    1.01
                                  ],
                                  "nested_str": "foo"
                                }
                              }
                            }
                          ]
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW",
                          "outputs": [
                            {
                              "src": "principal.terry_tibbs.vdefault#create-rule",
                              "val": [
                                "foo",
                                [
                                  "bar",
                                  true
                                ]
                              ]
                            }
                          ]
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "John's request without lenientScopeSearch",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "john_equipment_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    }
                  ]
                },
                {
                  "name": "john_pending_equipment_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Tests for equipment requests"
    }
  ],
  "summary": {
    "overallResult": "RESULT_PASSED",
    "testsCount": 24,
    "resultCounts": [
      {
        "result": "RESULT_PASSED",
        "count": 24
      }
    ]
  }
}
